qfieldsync/gui/cloud_login_dialog.py

```diff
@@ -195,5 +350,17 @@ def on_login_finished(self) -> None:
         self.rememberMeCheckBox.setEnabled(False)
         self.done(QDialog.Accepted)
 
+    def on_login_with_sso_provider_button_clicked(self, provider_data: dict) -> None:
```

Comment:
```suggestion
    def on_login_with_sso_provider_button_clicked(self, provider_data: Dict[..., ...]) -> None:
```
Missing key and value types for the dict. I guess `Dict[str, Any]`?